app/policies/post_policy.rb

```diff
@@ -0,0 +1,23 @@
+class PostPolicy < ApplicationPolicy
+  def show?
+    user.present? || record.status == 'public'
```

Comment:
Should be "(post public?) OR (user's own post?)" in my understanding. Right now any logged in users can see all the posts regardless of their status.
Also, take advantage of readable enum methods like
```
record.public?
```
If your model does not have them - create them.